src/arch/xtensa/smp/xtos/reset-vector.S

```diff
@@ -335,8 +335,18 @@ _ResetHandler:
 
 #if XCHAL_HAVE_PREFETCH
 	/* Enable cache prefetch if present.  */
-	movi	a2, XCHAL_CACHE_PREFCTL_DEFAULT
-	wsr.prefctl	a2
+#if defined(CONFIG_APOLLOLAKE)
+
+#if defined(CONFIG_SKYLAKE) || defined(CONFIG_KABYLAKE)
+	movi.n	a2, 0	/* skylake and kabylake */
+#else
+	movi.n	a2, 34  /* apollolake */
+#endif
```

Comment:
same comment as the other PR, nesting SKL and KBL under APL isn't very nice. Folks will screw-up...

Reply:
I'm going to fix this after @jajanusz has merged the build update, but in the mean time this is probably needed for improved runtime stability/performance.